### PR TITLE
docs / e2e: Update testing.md and remove the old e2e directory

### LINKED ISSUE #1003: update testing.md
From https://github.com/kubernetes/kops/issues/988#issuecomment-263345566

@zmerlynn said he needed to update testing.md, can you also move it to a folder?  dev??

### PR DESCRIPTION
**Pull request by zmerlynn**
Fixes #1003

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1300)

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -44,12 +44,81 @@ spec:
...
-The [e2e](../e2e/README.md) directory has a docker image and some scripts which make it easy to run
-the kubernetes e2e tests, using kops.
+The simple way:
+
+```
+# cd wherever you tend to put git repos
+git clone https://github.com/kubernetes/test-infra.git
+export KOPS_E2E_STATE_STORE=s3://your-kops-state-store # Change to your state store path
+export KOPS_E2E_CLUSTER_NAME=e2e.cluster.name          # Change to an FQDN for your e2e cluster name
+test-infra/jobs/ci-kubernetes-e2e-kops-aws.sh |& tee /tmp/testlog
```
Comment (chrislovecnm) [inline docs/testing.md]: We probably need to mention that aws creds will be used and kops needs to be installed

Comment (chrislovecnm) [inline docs/testing.md]: So this builds kops from master, it builds it locally?

Comment (zmerlynn) [inline docs/testing.md]: > We probably need to mention that aws creds will be used and kops needs to be installed

Will fix.

> So this builds kops from master, it builds it locally?

No. As I tried to explain below, this is using a `kops` built from `master` (the latest CI build). It doesn't build it on demand. That's what the next section deals with.

Comment (zmerlynn) [inline docs/testing.md]: PTAL

**Comment by chrislovecnm** [comment]
@zmerlynn do we need a review from @kris-nova and @justinsb?  You requested there presence?

**Comment by justinsb** [comment]
LGTM - merging!

**Comment by zmerlynn** [comment]
@chrislovecnm: Nah, I was casting a net for a review. Glad two of you looked at it!

**Pull request merged**